### ISSUE TITLE
AuthComponent not fully executing

### ISSUE THREAD

**Issue by bravo-kernel** [dereuromark/cakephp-tinyauth#50]
While checking why my `auth_allow.ini` was not being processed I noticed the AuthComponent does not run anything after [this line](https://github.com/dereuromark/cakephp-tinyauth/blob/master/src/Controller/Component/AuthComponent.php#L46).

Might very well be caused by my setup but the AuthComponent `startup` and `beforeRender` methods are never executed. Looks like jwt-auth somehow preceeds it (throwing a 401 error).

```
    protected function _setupAuth()
    {
        $this->_controller->loadComponent('TinyAuth.Auth', [
            'storage' => 'Memory', // do not use session
            'checkAuthIn' => 'Controller.initialize', // make user data available in beforeFilter()
            'authenticate' => [
                'Form' => [
                    'fields' => ['username' => 'email', 'password' => 'password']
                ],
                'ADmad/JwtAuth.Jwt' => [
                    'finder' => 'obfuscated',
                    'userModel' => 'Users',
                    'fields' => [
                        'username' => 'id'
                    ],
                    'parameter' => 'token',
                    'queryDatasource' => true // must be true, see docblock
                ]
            ],
            'authorize' => [
                'TinyAuth.Tiny' => [
                    'multiRole' => true,
                    'rolesTable' => 'UserRoles',
                    'roleColumn' => Configure::readOrFail('TinyAuth.roleColumn'),
                    'allowUser' => false,
                    'authorizeByPrefix' => false,
                    'superAdminRole' => null,
                    'autoClearCache' => Configure::readOrFail('TinyAuth.clearCache')
                ]
            ],
            'unauthorizedRedirect' => false,
            'loginAction' => [
                'controller' => 'Users',
                'action' => 'login',
                'prefix' => false // make sub-namespaces use main logic (DRY)
            ]
        ]);
    }
```

**Comment by dereuromark** [comment]
Isn't startup also in the implemented event list?
And then it should run before the JwtAuth startup ideally.

**Comment by bravo-kernel** [comment]
Startup is indeed in my `parent::implementedEvents()`, see dump below.

```php
Array
(
    [Controller.initialize] => authCheck
    [Controller.startup] => startup
)
```

FWIW jwt-auth does not have a startup method.

**Comment by dereuromark** [comment]
So is jwt executing too much

**Comment by bravo-kernel** [comment]
I took a closer look at `AuthComponent` and IMHO there are a couple of things that should be changed:

1.  the `startup` event should be renamed to `authCheck` event: this solves the not-processing of `auth_allow_ini`
2. the `beforeRender` event should return parent: this solves data handled by Controller `beforeRender` not being processed
3. call to [`$this->user()`](https://github.com/dereuromark/cakephp-tinyauth/blob/master/src/Controller/Component/AuthComponent.php#L69) in `beforeRender` causes *"Maximum function nesting level of '100' reached, aborting!"*. Not sure which method is being called but it's not the one in [AuthUserTrait](https://github.com/dereuromark/cakephp-tinyauth/blob/master/src/Auth/AuthUserTrait.php#L71) since the trait is not used by the component
4. `autoClearCache` option should not be coupled with `debug` setting, took me time to figure out wtf was happening in non-debug mode. Users should be able to use debug setting as autoClearCache setting (just as with TinyAuthorize)

I can provide a fix for all points except number 3. I really don't understand what's happening there.

**Comment by dereuromark** [comment]
Ok. Please do.
I will Take a closer look

**Comment by dereuromark** [comment]
" "Maximum function nesting level of '100' reached, "
How is that possible if there is no such nesting? user() is in parent, but there is no other method like this, So why should it re-execute the callback?

**Comment by bravo-kernel** [comment]
Not sure if I already explained but... that error does not occur anymore. Things are all working for me now as expected.

**Comment by dereuromark** [comment]
Very strange indeed.

**Comment by bravo-kernel** [comment]
closes through #51
